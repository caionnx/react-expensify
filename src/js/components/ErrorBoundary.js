import React from 'react'
import PropTypes from 'prop-types'

class ErrorBoundary extends React.Component {
  state = { error: null }

  componentDidCatch (error) {
    this.setState({ error })
  }

  render () {
    const { error } = this.state

    if (!error) {
      return this.props.children
    }

    return (
      <div className='content-container'>
        <h2>Something went wrong loading this page. Please try to reload the app.</h2>
        <details>
          <summary>Details for nerds</summary>
          {error.toString && error.toString()}
        </details>
      </div>
    )
  }
}

ErrorBoundary.propTypes = {
  children: PropTypes.any
}

export default ErrorBoundary
