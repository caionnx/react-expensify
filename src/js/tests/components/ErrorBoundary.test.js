import React from 'react'
import { shallow } from 'enzyme'
import ErrorBoundary from '../../components/ErrorBoundary'

let wrapper
const Component = () => null

test('should render the children component correctly', () => {
  wrapper = shallow(
    <ErrorBoundary>
      <Component />
    </ErrorBoundary>
  )

  expect(wrapper.find(Component)).toHaveLength(1)
})

test('should render an error message if the child component produces an error', () => {
  wrapper = shallow(
    <ErrorBoundary>
      <Component />
    </ErrorBoundary>
  )
  wrapper.find(Component).simulateError(new Error('Ops'))
  expect(wrapper).toMatchSnapshot()
})
