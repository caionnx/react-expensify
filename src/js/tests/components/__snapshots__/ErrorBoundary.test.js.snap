// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an error message if the child component produces an error 1`] = `
<div
  className="content-container"
>
  <h2>
    Something went wrong loading this page. Please try to reload the app.
  </h2>
  <details>
    <summary>
      Details for nerds
    </summary>
    Error: Ops
  </details>
</div>
`;
